consensus/hotstuff/vote_collector.go

```diff
@@ -23,17 +26,25 @@ const (
 	VoteCollectorStatusInvalid
 )
 
-// VoteCollector collects votes for the same block, produces QC when enough votes are collected
-// VoteCollector takes a callback function to report the event that a QC has been produced.
-type VoteCollector interface {
+func (ps VoteCollectorStatus) String() string {
+	names := [...]string{"VoteCollectorStatusCaching",
```

Comment:
Each call would construct this slice, would it be faster if moving it outside of this function and reused?

Reply:
Basically took what was implemented in approvals, yeah, makes sense to move it out.